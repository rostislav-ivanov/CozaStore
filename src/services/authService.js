const baseUrl = "https://cozastore-server.onrender.com/users";

export async function login({ email, password }) {
  const user = {
    email,
    password,
  };
  const response = await fetch(`${baseUrl}/login`, {
    method: "post",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(user),
  });
  return response.json();
}

export async function register({ email, password }) {
  const user = {
    email,
    password,
  };
  const response = await fetch(`${baseUrl}/register`, {
    method: "post",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(user),
  });
  return response.json();
}

export async function logout() {
  const auth = localStorage.getItem("auth");
  const authObj = JSON.parse(auth);
  const accessToken = authObj.accessToken;
  const response = await fetch(`${baseUrl}/logout`, {
    method: "get",
    headers: { "X-Authorization": accessToken },
  });
  return response.status === 204 ? "ok" : "error";
}
