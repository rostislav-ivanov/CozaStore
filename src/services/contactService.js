const baseUrl = "http://localhost:3030/jsonstore";

export async function sendMessage(message) {
  try {
    const response = await fetch(`${baseUrl}/messages`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(message),
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();
    return data;
  } catch (error) {
    throw new Error(`Failed to send message: ${error.message}`);
  }
}

export async function subscribe(email) {
  try {
    const response = await fetch(`${baseUrl}/subscribe`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ email }),
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();

    return data;
  } catch (error) {
    throw new Error(`Failed to subscribe: ${error.message}`);
  }
}
