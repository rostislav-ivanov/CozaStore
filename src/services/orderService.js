const baseUrl = "https://cozastore-server.onrender.com/data/orders";

export async function createOrder(order) {
  try {
    const auth = JSON.parse(localStorage.getItem("auth"));
    if (!auth || !auth.accessToken) {
      throw new Error("User is not logged in");
    }
    const response = await fetch(baseUrl, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "X-Authorization": auth.accessToken,
      },
      body: JSON.stringify(order),
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();
    return data;
  } catch (error) {
    throw new Error(`Failed to fetch orders: ${error.message}`);
  }
}

export async function getOrders() {
  try {
    const auth = JSON.parse(localStorage.getItem("auth"));
    if (!auth || !auth.accessToken) {
      throw new Error("User is not logged in");
    }

    const query = new URLSearchParams({
      where: `_ownerId="${auth._id}"`,
      sortBy: "_createdOn desc",
    });

    const queryString = query.toString().replace(/\+/g, "%20");

    const response = await fetch(`${baseUrl}?${queryString}`, {
      headers: {
        "X-Authorization": auth.accessToken,
      },
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();
    return data;
  } catch (error) {
    throw new Error(`Failed to fetch orders: ${error.message}`);
  }
}
