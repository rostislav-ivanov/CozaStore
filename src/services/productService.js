const baseUrl = `${import.meta.env.VITE_BASE_URL}/api/products`;

export async function getAllProducts(category, page = 1, pageSize = 6) {
  const offset = (page - 1) * pageSize;

  const query = new URLSearchParams({
    offset,
    pageSize,
    category,
  });

  try {
    const response = await fetch(`${baseUrl}?${query}`);

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();
    return Object.values(data);
  } catch (error) {
    throw new Error(error.message);
  }
}

export async function getProductsCount(category) {
  const query = new URLSearchParams({
    category,
  });

  try {
    const response = await fetch(`${baseUrl}/count?${query}`);

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();
    return data;
  } catch (error) {
    throw new Error(error.message);
  }
}

export async function getProductById(id) {
  const response = await fetch(`${baseUrl}/${id}/quick`);
  const data = await response.json();
  return data;
}

export async function getDetailsById(id) {
  const response = await fetch(`${baseUrl}/${id}`);
  const data = await response.json();
  return data;
}

export async function getWishListProducts() {
  try {
    const auth = JSON.parse(localStorage.getItem("auth"));
    if (!auth || !auth.accessToken || !auth.id) {
      throw new Error("User is not logged in");
    }

    const response = await fetch(`${baseUrl}/user`, {
      method: "GET",
      credentials: "include",
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();
    return Object.values(data);
  } catch (error) {
    throw new Error(`Failed to get wishes: ${error.message}`);
  }
}
