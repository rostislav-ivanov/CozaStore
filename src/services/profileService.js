const baseUrl = "https://cozastore-server.onrender.com/data/profiles";

export async function createProfile(auth) {
  if (!auth || !auth.accessToken) {
    throw new Error("User is not logged in");
  }

  try {
    const response = await fetch(`${baseUrl}`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "X-Authorization": auth.accessToken,
      },
      body: JSON.stringify({
        email: auth.email,
        firstName: "",
        lastName: "",
        phone: "",
        shippingCity: "",
        shippingOffice: "",
      }),
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    return response.json();
  } catch (error) {
    throw new Error(`Failed to extend user: ${error.message}`);
  }
}

export async function getProfile() {
  try {
    const auth = JSON.parse(localStorage.getItem("auth"));
    if (!auth || !auth.accessToken || !auth._id) {
      throw new Error("User is not logged in");
    }

    const query = new URLSearchParams({
      where: `_ownerId="${auth._id}"`,
    });

    const response = await fetch(`${baseUrl}?${query}`, {
      method: "GET",
      headers: {
        "X-Authorization": auth.accessToken,
      },
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();
    if (data.length < 1) {
      throw new Error("User not found");
    }
    return data[0];
  } catch (error) {
    throw new Error(`Failed to get user: ${error.message}`);
  }
}

export async function updateProfile(profile) {
  try {
    const auth = JSON.parse(localStorage.getItem("auth"));
    if (!auth || !auth.accessToken) {
      throw new Error("User is not logged in");
    }

    const response = await fetch(`${baseUrl}/${profile._id}`, {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
        "X-Authorization": auth.accessToken,
      },
      body: JSON.stringify({ ...profile }),
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    return response.json();
  } catch (error) {
    throw new Error(`Failed to add wish: ${error.message}`);
  }
}
