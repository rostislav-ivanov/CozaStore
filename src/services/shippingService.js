const baseUrl = `${import.meta.env.VITE_BASE_URL}/api/econt`;

export async function getCities() {
  try {
    const response = await fetch(`${baseUrl}/cities`);
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }
    const data = await response.json();
    const cities = data.sort((a, b) => a.name.localeCompare(b.name));
    return cities;
  } catch (error) {
    console.error("Error fetching data:", error);
  }
}

export async function getOffices(cityID) {
  try {
    const response = await fetch(`${baseUrl}/offices/${cityID}`);
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }
    const data = await response.json();
    const offices = data.sort((a, b) => a.name.localeCompare(b.name));
    return offices;
  } catch (error) {
    console.error("Error fetching data:", error);
  }
}
