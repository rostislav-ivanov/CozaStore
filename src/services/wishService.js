const baseUrl = `${import.meta.env.VITE_BASE_URL}/api/wishes`;

export async function getWish() {
  try {
    const response = await fetch(baseUrl, {
      method: "GET",
      credentials: "include",
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    let data = await response.json();

    const wishes = data.reduce((acc, wishId) => {
      acc[wishId] = true;
      return acc;
    }, {});

    return wishes;
  } catch (error) {
    throw new Error(`Failed to get wishes list: ${error.message}`);
  }
}

export async function updateWish(wishData) {
  try {
    const auth = JSON.parse(localStorage.getItem("auth"));
    if (!auth || !auth.accessToken) {
      throw new Error("User is not logged in");
    }
    const response = await fetch(baseUrl, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(Object.keys(wishData)),
      credentials: "include",
    });

    if (!response.ok) {
      throw new Error(`Error: ${response.status} ${response.statusText}`);
    }

    return response.json();
  } catch (error) {
    throw new Error(`Failed to add wishes: ${error.message}`);
  }
}
