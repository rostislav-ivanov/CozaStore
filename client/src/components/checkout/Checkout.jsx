import { useContext, useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";

import { BagContext } from "../../context/bagContext";
import styles from "./Checkout.module.css";
import * as shippingService from "../../services/shippingService";
import * as orderService from "../../services/orderService";

export default function Checkout() {
  const { bag, removeItem, updateItem, clearBag } = useContext(BagContext);
  const subTotal = bag.reduce(
    (acc, item) => acc + item.price * item.quantity,
    0
  );

  const [cities, setCities] = useState([]);
  const [city, setCity] = useState("");
  const [offices, setOffices] = useState([]);
  const [office, setOffice] = useState("");
  const [shippingPrice, setShippingPrice] = useState("");
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [phone, setPhone] = useState("");
  const [errors, setErrors] = useState({});
  const [loading, setLoading] = useState(false);
  const navigate = useNavigate();

  useEffect(() => {
    setLoading(true);
    shippingService.getCities().then((data) => {
      setCities(data);
      setLoading(false);
    });
  }, []);

  useEffect(() => {
    if (!city.id) {
      return;
    }
    setLoading(true);
    shippingService.getOffices(city.id).then((data) => {
      setOffices(data);
      setLoading(false);
    });
  }, [city]);

  const cityHandler = (e) => {
    const id = e.target.options[e.target.selectedIndex].id;
    const name = e.target.value;
    setCity({ id, name });
    setShippingPrice("");
    setErrors({ ...errors, city: "" });
  };

  const officeHandler = (e) => {
    const id = e.target.options[e.target.selectedIndex].id;
    const name = e.target.value;
    setOffice({ id, name });
    setErrors({ ...errors, office: "" });
    setShippingPrice(4.99);
  };

  const validateInputs = () => {
    const newErrors = {};
    if (!city) {
      newErrors.city = "City is required.";
    }
    if (!office) {
      newErrors.office = "Office is required.";
    }
    if (!firstName) {
      newErrors.firstName = "First name is required.";
    }
    if (!lastName) {
      newErrors.lastName = "Last name is required.";
    }
    if (!phone) {
      newErrors.phone = "Phone is required.";
    }
    setErrors(newErrors);
    return Object.keys(newErrors).length === 0;
  };

  const submitHandler = async (e) => {
    e.preventDefault();
    if (!validateInputs()) {
      return;
    }
    const order = {
      status: "Unfulfilled",
      city: city.name,
      office: office.name,
      firstName,
      lastName,
      phone,
      products: bag,
      shippingPrice,
      total: subTotal + shippingPrice,
    };
    try {
      const result = await orderService.createOrder(order);
      clearBag();
      alert(
        `Order placed successfully! Your order number is #${result._createdOn}`
      );
      navigate("/");
    } catch (error) {
      alert(error.message);
      return;
    }
  };

  return (
    <form
      className="bg0 p-t-75 p-b-85"
      onSubmit={(e) => {
        submitHandler(e);
      }}
    >
      <div className="container">
        <div className="row">
          <div className="col-lg-10 col-xl-7 m-lr-auto m-b-50">
            <div className="m-l-25 m-r--38 m-lr-0-xl">
              <div className="wrap-table-shopping-cart">
                <table className="table-shopping-cart">
                  <thead>
                    <tr className="table_head">
                      <th className="column-1">Product</th>
                      <th className="column-2"></th>
                      <th className="column-3">Price</th>
                      <th className="column-4">Quantity</th>
                      <th className="column-5">Total</th>
                    </tr>
                  </thead>
                  <tbody>
                    {bag.map((item, index) => (
                      <tr key={index} className="table_row">
                        <td className="column-1">
                          <div className="how-itemcart1">
                            <img src={item.image} alt="IMG" />
                          </div>
                        </td>
                        <td className="column-2">
                          <div className="row">
                            <span> {item.name}</span>
                            <span>Color: {item.color}</span>
                            <span>Size: {item.size}</span>
                          </div>
                        </td>
                        <td className="column-3">$ {item.price.toFixed(2)}</td>
                        <td className="column-4">
                          <div className="wrap-num-product flex-w m-l-auto m-r-0">
                            <div className="btn-num-product-down cl8 hov-btn3 trans-04 flex-c-m">
                              <i
                                className="fs-16 zmdi zmdi-minus"
                                onClick={() =>
                                  updateItem(index, item.quantity - 1)
                                }
                              ></i>
                            </div>

                            <input
                              className="mtext-104 cl3 txt-center num-product"
                              type="number"
                              name="quantity"
                              value={Number(item.quantity)}
                              onChange={(e) =>
                                updateItem(index, Number(e.target.value))
                              }
                            />

                            <div className="btn-num-product-up cl8 hov-btn3 trans-04 flex-c-m">
                              <i
                                className="fs-16 zmdi zmdi-plus"
                                onClick={() =>
                                  updateItem(index, item.quantity + 1)
                                }
                              ></i>
                            </div>
                          </div>
                        </td>
                        <td className="column-5">
                          ${" "}
                          {(bag[index].price * bag[index].quantity).toFixed(2)}
                          <i
                            style={{ cursor: "pointer" }}
                            className="fs-16 hov-cl1 p-l-20 zmdi zmdi-delete"
                            onClick={() => removeItem(index)}
                          ></i>
                        </td>
                      </tr>
                    ))}
                  </tbody>
                </table>
              </div>
            </div>
          </div>

          <div className="col-sm-10 col-lg-7 col-xl-5 m-lr-auto m-b-50">
            <div className="bor10 p-lr-40 p-t-30 p-b-40 m-l-63 m-r-40 m-lr-0-xl p-lr-15-sm">
              <h4 className="mtext-109 cl2 p-b-30">Cart Totals</h4>

              <div className="flex-w flex-t bor12 p-b-13">
                <div className="size-208">
                  <span className="stext-110 cl2">Subtotal:</span>
                </div>

                <div className="size-209">
                  <span className="mtext-110 cl2">$ {subTotal.toFixed(2)}</span>
                </div>
              </div>

              <div className="flex-w flex-t bor12 p-t-15 p-b-30">
                <div className="size-208 w-full-ssm">
                  <span className="stext-110 cl2">Shipping:</span>
                </div>

                <div className="size-209 p-r-18 p-r-0-sm w-full-ssm">
                  {shippingPrice != "" && (
                    <div className="size-209">
                      <span className="mtext-110 cl2">
                        $ {Number(shippingPrice).toFixed(2)}
                      </span>
                    </div>
                  )}
                  {!shippingPrice && (
                    <p className="stext-111 cl6 p-t-2">
                      There are no shipping address selected. We ship only to
                      Bulgaria by ECONT courier. Please select city and office
                      of ECONT, or contact us if you need any help.
                    </p>
                  )}
                  <div className="p-t-15">
                    {loading && <p>Loading...</p>}
                    <div className="p-tb-10" hidden={loading}>
                      {errors.city && (
                        <div className={styles.error}>{errors.city}</div>
                      )}
                      <select
                        className="stext-111 cl8 plh3 size-111 p-lr-15"
                        name="city"
                        id="city"
                        onChange={cityHandler}
                      >
                        <option value="">Choose a city</option>
                        {cities.length > 0 &&
                          cities.map((city) => (
                            <option
                              key={city.id}
                              id={city.id}
                              value={city.name}
                            >
                              {city.name}
                            </option>
                          ))}
                      </select>
                    </div>

                    {city && (
                      <div className="p-tb-10" hidden={loading}>
                        {errors.office && (
                          <div className={styles.error}>{errors.office}</div>
                        )}
                        {offices.length <= 0 && (
                          <p className="stext-111 cl6 p-t-2">
                            There are no offices available in{" "}
                            <strong>{city.name}</strong>. Please check other
                            city, or contact us if you need any help.
                          </p>
                        )}
                        {offices.length > 0 && (
                          <select
                            className="stext-111 cl8 plh3 size-111 p-lr-15"
                            name="office"
                            id="office"
                            onChange={officeHandler}
                          >
                            <option value="">Choose a office</option>
                            {offices.length > 0 &&
                              offices.map((office) => (
                                <option
                                  key={office.id}
                                  id={office.id}
                                  value={office.name}
                                >
                                  {office.name}
                                </option>
                              ))}
                          </select>
                        )}
                      </div>
                    )}
                    {errors.firstName && (
                      <div className={styles.error}>{errors.firstName}</div>
                    )}

                    <div className="bor8 bg0 m-tb-10">
                      <input
                        className="stext-111 cl8 plh3 size-111 p-lr-15"
                        type="text"
                        name="firstName"
                        placeholder="First Name"
                        value={firstName}
                        onChange={(e) => {
                          const firstName = e.target.value;
                          setFirstName(firstName);
                          if (firstName.length > 0) {
                            setErrors({ ...errors, firstName: "" });
                          } else {
                            setErrors({
                              ...errors,
                              firstName: "First name is required.",
                            });
                          }
                        }}
                      />
                    </div>
                    {errors.lastName && (
                      <div className={styles.error}>{errors.lastName}</div>
                    )}
                    <div className="bor8 bg0 m-tb-10">
                      <input
                        className="stext-111 cl8 plh3 size-111 p-lr-15"
                        type="text"
                        name="lastName"
                        placeholder="Last Name"
                        value={lastName}
                        onChange={(e) => {
                          const lastName = e.target.value;
                          setLastName(lastName);
                          if (lastName.length > 0) {
                            setErrors({ ...errors, lastName: "" });
                          } else {
                            setErrors({
                              ...errors,
                              lastName: "Last name is required.",
                            });
                          }
                        }}
                      />
                    </div>
                    {errors.phone && (
                      <div className={styles.error}>{errors.phone}</div>
                    )}
                    <div className="bor8 bg0 m-tb-10">
                      <input
                        className="stext-111 cl8 plh3 size-111 p-lr-15"
                        type="text"
                        name="phone"
                        placeholder="Phone"
                        value={phone}
                        onChange={(e) => {
                          const phone = e.target.value;
                          setPhone(phone);
                          if (phone.length > 0) {
                            setErrors({ ...errors, phone: "" });
                          } else {
                            setErrors({
                              ...errors,
                              phone: "Phone is required.",
                            });
                          }
                        }}
                      />
                    </div>
                  </div>
                </div>
              </div>

              <div className="flex-w flex-t p-t-27 p-b-33">
                <div className="size-208">
                  <span className="mtext-101 cl2">Total:</span>
                </div>

                <div className="size-209 p-t-1">
                  <span className="mtext-110 cl2">
                    $ {Number(subTotal + shippingPrice).toFixed(2)}
                  </span>
                </div>
              </div>

              <button className="flex-c-m stext-101 cl0 size-116 bg3 bor14 hov-btn3 p-lr-15 trans-04 pointer">
                Proceed to Checkout
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  );
}
