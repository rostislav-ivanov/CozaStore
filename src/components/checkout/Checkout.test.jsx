import { test, describe, expect, beforeEach } from "vitest";
import {
  render,
  screen,
  prettyDOM,
  fireEvent,
  waitFor,
  getByText,
} from "../../tests/test-utils";

import Checkout from "./Checkout";
import * as profileService from "../../services/profileService";
import * as shippingService from "../../services/shippingService";
import {
  mockBagContextValue,
  mockBagDate,
  mockCount,
  mockProfileData,
  mockCitiesData,
  mockOfficesData,
} from "../../tests/mockData";

describe("Checkout component", () => {
  beforeEach(() => {
    mockBagContextValue.bag = [...mockBagDate];
    mockBagContextValue.count = mockCount;
  });

  test("renders correctly", () => {
    const { container } = render(<Checkout />);
    //console.log(prettyDOM(container));
    expect(container).toMatchSnapshot();
  });

  test("displays items in the cart", () => {
    render(<Checkout />);
    const imgElements = [...screen.getAllByAltText("IMG")];
    const quantities = [...screen.getAllByRole("spinbutton")];
    mockBagDate.forEach((item, index) => {
      expect(quantities[index]).toHaveValue(item.quantity);
      expect(imgElements[index]).toHaveAttribute("src", item.image);
      expect(screen.getByText(`Color: ${item.color}`)).toBeInTheDocument();
      expect(screen.getByText(`Size: ${item.size}`)).toBeInTheDocument();
      expect(screen.getByText(item.name)).toBeInTheDocument();
      expect(
        screen.getByText(`$ ${Number(item.price).toFixed(2)}`)
      ).toBeInTheDocument();
    });
  });

  test("display subtotal", () => {
    const subTotal = mockBagContextValue.bag.reduce(
      (acc, item) => acc + item.price * item.quantity,
      0
    );
    render(<Checkout />);
    const subtotalLabel = screen.getByText("Subtotal:");
    const parentElement = subtotalLabel.closest(".flex-w.flex-t.bor12.p-b-13");
    const subtotalValue = parentElement.children[1];
    expect(subtotalValue).toHaveTextContent(`$ ${subTotal.toFixed(2)}`);
  });

  test("display shipping address if it is in user profile", async () => {
    profileService.getProfile.mockResolvedValue(mockProfileData);
    shippingService.getCities.mockResolvedValue(mockCitiesData);
    shippingService.getOffices.mockResolvedValue(mockOfficesData);
    render(<Checkout />);
    await waitFor(
      () => expect(profileService.getProfile).toHaveBeenCalled(),
      expect(shippingService.getCities).toHaveBeenCalled()
    );
    await waitFor(() =>
      expect(
        screen.queryByText(/There are no offices available in/)
      ).not.toBeInTheDocument()
    );
    const firstNameInput = screen.getByPlaceholderText("First Name");
    expect(firstNameInput).toHaveValue(mockProfileData.firstName);
    const lastNameInput = screen.getByPlaceholderText("Last Name");
    expect(lastNameInput).toHaveValue(mockProfileData.lastName);
    const phoneInput = screen.getByPlaceholderText("Phone");
    expect(phoneInput).toHaveValue(mockProfileData.phone);
    const cityOption = document.getElementById("city");
    expect(cityOption.value).toBe(mockProfileData.shippingCity);
    const officeOption = document.getElementById("office");
    expect(officeOption.value).toBe(mockProfileData.shippingOffice);
  });
});
