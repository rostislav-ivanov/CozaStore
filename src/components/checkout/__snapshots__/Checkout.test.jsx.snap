// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkout component > renders correctly 1`] = `
<div>
  <form
    class="bg0 p-t-75 p-b-85"
  >
    <div
      class="_spinnerContainer_d6a07c"
    >
      <div
        class="spinner-border"
        role="status"
      >
        <span
          class="visually-hidden"
        >
          Loading...
        </span>
      </div>
    </div>
    <div
      class="container loadingOverlay"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-10 col-xl-7 m-lr-auto m-b-50"
        >
          <div
            class="m-l-25 m-r--38 m-lr-0-xl"
          >
            <div
              class="wrap-table-shopping-cart"
            >
              <table
                class="table-shopping-cart"
              >
                <thead>
                  <tr
                    class="table_head"
                  >
                    <th
                      class="column-1"
                    >
                      Product
                    </th>
                    <th
                      class="column-2"
                    />
                    <th
                      class="column-3"
                    >
                      Price
                    </th>
                    <th
                      class="column-4"
                    >
                      Quantity
                    </th>
                    <th
                      class="column-5"
                    >
                      Total
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    class="table_row"
                  >
                    <td
                      class="column-1"
                    >
                      <div
                        class="how-itemcart1"
                      >
                        <img
                          alt="IMG"
                          src="/images/product-01.jpg"
                        />
                      </div>
                    </td>
                    <td
                      class="column-2"
                    >
                      <div
                        class="row"
                      >
                        <span>
                           
                          Esprit Ruffle Shirt
                        </span>
                        <span>
                          Color: 
                          Red
                        </span>
                        <span>
                          Size: 
                          M
                        </span>
                      </div>
                    </td>
                    <td
                      class="column-3"
                    >
                      $ 
                      16.64
                    </td>
                    <td
                      class="column-4"
                    >
                      <div
                        class="wrap-num-product flex-w m-l-auto m-r-0"
                      >
                        <div
                          class="btn-num-product-down cl8 hov-btn3 trans-04 flex-c-m"
                        >
                          <i
                            class="fs-16 zmdi zmdi-minus"
                          />
                        </div>
                        <input
                          class="mtext-104 cl3 txt-center num-product"
                          name="quantity"
                          type="number"
                          value="2"
                        />
                        <div
                          class="btn-num-product-up cl8 hov-btn3 trans-04 flex-c-m"
                        >
                          <i
                            class="fs-16 zmdi zmdi-plus"
                          />
                        </div>
                      </div>
                    </td>
                    <td
                      class="column-5"
                    >
                      $
                       
                      33.28
                      <i
                        class="fs-16 hov-cl1 p-l-20 zmdi zmdi-delete"
                        style="cursor: pointer;"
                      />
                    </td>
                  </tr>
                  <tr
                    class="table_row"
                  >
                    <td
                      class="column-1"
                    >
                      <div
                        class="how-itemcart1"
                      >
                        <img
                          alt="IMG"
                          src="/images/product-03.jpg"
                        />
                      </div>
                    </td>
                    <td
                      class="column-2"
                    >
                      <div
                        class="row"
                      >
                        <span>
                           
                          Only Check Trouser
                        </span>
                        <span>
                          Color: 
                          Blue
                        </span>
                        <span>
                          Size: 
                          S
                        </span>
                      </div>
                    </td>
                    <td
                      class="column-3"
                    >
                      $ 
                      25.50
                    </td>
                    <td
                      class="column-4"
                    >
                      <div
                        class="wrap-num-product flex-w m-l-auto m-r-0"
                      >
                        <div
                          class="btn-num-product-down cl8 hov-btn3 trans-04 flex-c-m"
                        >
                          <i
                            class="fs-16 zmdi zmdi-minus"
                          />
                        </div>
                        <input
                          class="mtext-104 cl3 txt-center num-product"
                          name="quantity"
                          type="number"
                          value="3"
                        />
                        <div
                          class="btn-num-product-up cl8 hov-btn3 trans-04 flex-c-m"
                        >
                          <i
                            class="fs-16 zmdi zmdi-plus"
                          />
                        </div>
                      </div>
                    </td>
                    <td
                      class="column-5"
                    >
                      $
                       
                      76.50
                      <i
                        class="fs-16 hov-cl1 p-l-20 zmdi zmdi-delete"
                        style="cursor: pointer;"
                      />
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          class="col-sm-10 col-lg-7 col-xl-5 m-lr-auto m-b-50"
        >
          <div
            class="bor10 p-lr-40 p-t-30 p-b-40 m-l-63 m-r-40 m-lr-0-xl p-lr-15-sm"
          >
            <h4
              class="mtext-109 cl2 p-b-30"
            >
              Cart Totals
            </h4>
            <div
              class="flex-w flex-t bor12 p-b-13"
            >
              <div
                class="size-208"
              >
                <span
                  class="stext-110 cl2"
                >
                  Subtotal:
                </span>
              </div>
              <div
                class="size-209"
              >
                <span
                  class="mtext-110 cl2"
                >
                  $ 
                  109.78
                </span>
              </div>
            </div>
            <div
              class="flex-w flex-t bor12 p-t-15 p-b-30"
            >
              <div
                class="size-208 w-full-ssm"
              >
                <span
                  class="stext-110 cl2"
                >
                  Shipping:
                </span>
              </div>
              <div
                class="size-209 p-r-18 p-r-0-sm w-full-ssm"
              >
                <p
                  class="stext-111 cl6 p-t-2"
                >
                  There are no shipping address selected. We ship only to Bulgaria by ECONT courier. Please select city and office of ECONT, or contact us if you need any help.
                </p>
                <div
                  class="p-t-15"
                >
                  <div
                    class="p-tb-10"
                  >
                    <select
                      class="stext-111 cl8 plh3 size-111 p-lr-15"
                      id="city"
                      name="city"
                    >
                      <option
                        disabled=""
                        value=""
                      >
                        Choose a city
                      </option>
                    </select>
                  </div>
                  <div
                    class="bor8 bg0 m-tb-10"
                  >
                    <input
                      class="stext-111 cl8 plh3 size-111 p-lr-15"
                      name="firstName"
                      placeholder="First Name"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="bor8 bg0 m-tb-10"
                  >
                    <input
                      class="stext-111 cl8 plh3 size-111 p-lr-15"
                      name="lastName"
                      placeholder="Last Name"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="bor8 bg0 m-tb-10"
                  >
                    <input
                      class="stext-111 cl8 plh3 size-111 p-lr-15"
                      name="phone"
                      placeholder="Phone"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex-w flex-t p-t-27 p-b-33"
            >
              <div
                class="size-208"
              >
                <span
                  class="mtext-101 cl2"
                >
                  Total:
                </span>
              </div>
              <div
                class="size-209 p-t-1"
              >
                <span
                  class="mtext-110 cl2"
                >
                  $ 
                  109.78
                </span>
              </div>
            </div>
            <button
              class="flex-c-m stext-101 cl0 size-116 bg3 bor14 hov-btn3 p-lr-15 trans-04 pointer"
            >
              Proceed to Checkout
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
